File: app/src/main/java/nl/rsdt/japp/jotial/maps/searching/Adapter.java
package nl.rsdt.japp.jotial.maps.searching;

import android.content.ContentProvider;
import android.content.ContentValues;
import android.database.Cursor;
import android.database.MatrixCursor;
import android.net.Uri;
import androidx.annotation.Nullable;

/**
 * @author Dingenis Sieger Sinke
 * @version 1.0
 * @since 12-9-2016
 * Description...
 */
public class Adapter extends ContentProvider {

    @Override
    public boolean onCreate() {
        return false;
    }

    @Nullable
    @Override
    public Cursor query(Uri uri, String[] strings, String s, String[] strings1, String s1) {
        MatrixCursor cursor = new MatrixCursor(new String[] { "marker_id", "value" } );
        cursor.addRow(new Object[] { } );
        return null;
    }

    @Nullable
    @Override
    public String getType(Uri uri) {
        return null;
    }

    @Nullable
    @Override
    public Uri insert(Uri uri, ContentValues contentValues) {
        return null;
    }

    @Override
    public int delete(Uri uri, String s, String[] strings) {
        return 0;
    }

    @Override
    public int update(Uri uri, ContentValues contentValues, String s, String[] strings) {
        return 0;
    }
}

File: app/src/main/java/nl/rsdt/japp/jotial/maps/searching/Adapter.kt
package nl.rsdt.japp.jotial.maps.searching

import android.content.ContentProvider
import android.content.ContentValues
import android.database.Cursor
import android.database.MatrixCursor
import android.net.Uri

/**
 * @author Dingenis Sieger Sinke
 * @version 1.0
 * @since 12-9-2016
 * Description...
 */
class Adapter : ContentProvider() {

    override fun onCreate(): Boolean {
        return false
    }

    override fun query(uri: Uri, strings: Array<String>?, s: String?, strings1: Array<String>?, s1: String?): Cursor? {
        val cursor = MatrixCursor(arrayOf("marker_id", "value"))
        cursor.addRow(arrayOf())
        return null
    }

    override fun getType(uri: Uri): String? {
        return null
    }

    override fun insert(uri: Uri, contentValues: ContentValues?): Uri? {
        return null
    }

    override fun delete(uri: Uri, s: String?, strings: Array<String>?): Int {
        return 0
    }

    override fun update(uri: Uri, contentValues: ContentValues?, s: String?, strings: Array<String>?): Int {
        return 0
    }
}
